image/api_selector.go

```diff
@@ -230,7 +236,15 @@ func (as *APISelector) buildCandidateTags(params *Params) []*tagSet {
 		sort.Strings(ts.Tags)
 	}
 
-	return result
+	for _, ts := range result {
+		for _, tag := range ts.Tags {
+			if strings.Contains(tag, ",") {
+				return result, workerErrors.NewWrappedJobAbortError(errors.Errorf("tag %v contained \",\", which is not supported by job-board -- check .travis.yml for trailing comma", tag))
```

Comment:
Is the intent that this error propagates up to the job output log?  If so, I think that this error can/should also be handled at a higher level prior to scheduling, such as at the gatekeeper level, so that an error message can be provided at the requests view and can report with richer context than "tag %v".  Even if this error string remains, I don't think it should contain any reference to `job-board`, or really to `.travis.yml`, but instead should be fairly low-level, optimized for backend debugging.